src/lib/annotations/Annotator.js

```diff
@@ -241,93 +240,116 @@ class Annotator extends EventEmitter {
+     * Toggles annotation modes on and off. When an annotation mode is
+     * on, annotation threads will be created at that location.
      *
+     * @param {string} mode - Current annotation mode
      * @param {HTMLEvent} event - DOM event
      * @return {void}
      */
-    togglePointAnnotationHandler(event = {}) {
+    toggleAnnotationHandler(mode, event = {}) {
         this.destroyPendingThreads();
-        const pointButtonSelector = this.modeButtons[TYPES.point].selector;
-        const buttonEl = event.target || this.previewUI.getAnnotateButton(pointButtonSelector);
 
-        if (this.isInDrawMode()) {
-            this.toggleDrawAnnotationHandler();
+        // No specific mode available for annotation type
+        if (!(mode in this.modeButtons)) {
+            return;
         }
 
-        // If in annotation mode, turn it off
-        if (this.isInPointMode()) {
-            this.emit('annotationmodeexit');
-            this.annotatedElement.classList.remove(CLASS_ANNOTATION_POINT_MODE);
-            if (buttonEl) {
-                buttonEl.classList.remove(CLASS_ACTIVE);
-            }
+        const buttonSelector = this.modeButtons[mode].selector;
+        const buttonEl = event.target || this.previewUI.getAnnotateButton(buttonSelector);
 
-            this.unbindModeListeners(); // Disable point mode
-            this.bindDOMListeners(); // Re-enable other annotations
+        // Exit any other annotation mode
+        this.exitAnnotationModes(mode, buttonEl);
 
-            // Otherwise, enable annotation mode
+        // If in annotation mode, turn it off
+        if (this.isInAnnotationMode(mode)) {
+            this.disableAnnotationMode(mode, buttonEl);
+
+            // Remove annotation mode
+            this.currentAnnotationMode = null;
         } else {
-            this.emit('annotationmodeenter', TYPES.point);
-            this.annotatedElement.classList.add(CLASS_ANNOTATION_POINT_MODE);
-            if (buttonEl) {
-                buttonEl.classList.add(CLASS_ACTIVE);
-            }
+            this.enableAnnotationMode(mode, buttonEl);
 
-            this.unbindDOMListeners(); // Disable other annotations
-            this.bindPointModeListeners(); // Enable point mode
+            // Update annotation mode
+            this.currentAnnotationMode = mode;
         }
     }
 
     /**
-     * Toggles draw annotation mode on and off. When draw annotation mode is
-     * on, a click and draw
+     * Disables the specified annotation mode
      *
-     * @param {HTMLEvent} event - DOM event
+     * @param {string} mode - Current annotation mode
+     * @param {HTMLElement} buttonEl - Annotation button element
      * @return {void}
      */
-    toggleDrawAnnotationHandler(event = {}) {
-        this.destroyPendingThreads();
-        if (this.isInPointMode()) {
-            this.togglePointAnnotationHandler();
+    disableAnnotationMode(mode, buttonEl) {
+        this.emit('annotationmodeexit');
```

Comment:
Could we move `annotationmodeexit`, "enter" and any other message strings to a constant we can reference around the code base? Or are they ever only used in one place, each?